src/liboslexec/shadingsys.cpp

```diff
@@ -2571,8 +2546,14 @@ ShadingContext *
 ShadingSystemImpl::get_context (PerThreadInfo *threadinfo,
                                 TextureSystem::Perthread *texture_threadinfo)
 {
-    if (! threadinfo)
-        threadinfo = get_perthread_info ();
+    if (! threadinfo) {
+#if OSL_VERSION < 20200
+        warning ("ShadingSystem::get_context called without a PerThreadInfo");
```

Comment:
Did you intend to auto-create a thread info here? otherwise it just crashes at the line below, no?